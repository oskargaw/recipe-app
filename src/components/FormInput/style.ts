import styled from "styled-components";
import { colors } from "../../theme/index";

export const StyledFormInput = styled.input`
  margin: 1rem 0;
  height: 2rem;
  border: none;
  border-radius: 0.3rem;
  background-color: ${colors.gray[100]};
  padding: 0.5rem 1rem;
  outline: none;
  font-family: sans-serif;
  &:focus {
    border: 0.1px solid ${colors.gray[200]};
  }
`;

export const FormInputError = styled.div`
  font-size: 0.8rem;
  color: ${colors.red[500]};
  margin-bottom: 1rem;
`;
