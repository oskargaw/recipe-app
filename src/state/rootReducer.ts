import { combineReducers } from "redux";
import { recipesReducer } from "./recipes/recipes.reducer";

const rootReducer = combineReducers({
  recipes: recipesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
