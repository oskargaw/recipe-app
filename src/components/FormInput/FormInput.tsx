import { UseFormRegister } from "react-hook-form";
import { FormLabel } from "../FormLabel";
import { FormInputError, StyledFormInput } from "./style";

export interface FormInputProps {
  name: string;
  type: string;
  register: UseFormRegister<any>;
  label: string;
  placeholder?: string;
  defaultValues?: object;
  error?: string;
}

export const FormInput: React.FC<FormInputProps> = ({
  name,
  type,
  label,
  placeholder,
  register,
  error,
}) => (
  <>
    <FormLabel htmlFor={name}>{label}</FormLabel>
    <StyledFormInput
      type={type}
      placeholder={placeholder}
      {...register(name)}
    />
    <FormInputError>{error}</FormInputError>
  </>
);
