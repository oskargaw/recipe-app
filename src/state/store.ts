import { createStore } from "redux";
import { persistReducer, persistStore } from "redux-persist"; // imports from redux-persist
import storage from "redux-persist/lib/storage";
import rootReducer from "./rootReducer";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(
  persistedReducer,
  (window as any).__REDUX_DEVTOOLS_EXTENSION__ &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION__()
);

const persistor = persistStore(store as any);

export { store, persistor };
