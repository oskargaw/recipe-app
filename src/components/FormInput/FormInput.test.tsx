import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { RootState } from "../../state/rootReducer";
import { FormInput, FormInputProps } from "./FormInput";

describe("<FormInput />", () => {
  const mockStore = configureStore([]);
  const initialState: RootState = {
    recipes: {
      recipeList: [],
    },
  };
  const store = mockStore(initialState);

  const mockRegister = jest.fn();

  const props: FormInputProps = {
    name: "test",
    type: "text",
    register: mockRegister,
    label: "test",
    placeholder: "test",
    error: "test",
  };

  it("should render correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <FormInput {...props} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display correct value after entering it inside the input", async () => {
    render(
      <Provider store={store}>
        <FormInput {...props} />
      </Provider>
    );

    const input = screen.getByPlaceholderText("test");
    expect(input).toBeInTheDocument();

    fireEvent.change(input, { target: { value: "banana bread" } });

    expect(input).toHaveValue("banana bread");
  });
});
