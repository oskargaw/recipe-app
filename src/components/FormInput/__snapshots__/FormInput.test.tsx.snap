// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> should render correctly 1`] = `
<DocumentFragment>
  <label
    class="sc-bdvvtL gJVqff"
    for="test"
  >
    test
  </label>
  <input
    class="sc-gsDKAQ dJjuOV"
    placeholder="test"
    type="text"
  />
  <div
    class="sc-dkPtRN dNQVQE"
  >
    test
  </div>
</DocumentFragment>
`;
